### PR TITLE
Model rerun file as a class and add tests - issue #865

### PR DESCRIPTION
**Pull request by mcrmfc**
Here is a starter for discussion around modelling the rerun file issue #865. See what you think and let me know. Cheers, Matt

### CODE REVIEW

File: lib/cucumber/cli/rerun_file.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module Cucumber
+  module Cli
+    class RerunFile
+      attr_reader :path
+
+      def self.rerun_file?(path)
```
Comment (mattwynne) [inline lib/cucumber/cli/rerun_file.rb]: if you change this to something more abstract like `#can_read?` then I think we could use this same pattern for the other feature-loading strategies, which would be nice.

**Comment by mattwynne** [comment]
Looks great to me!

**Comment by mcrmfc** [comment]
Thanks @mattwynne - changes made as suggested.

File: lib/cucumber/cli/rerun_file.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      def self.real_path(path)
+        path[1..-1] # remove leading @
+      end
+
+      def initialize(path)
+        @path = self.class.real_path(path)
+      end
+
+      def features
+        lines.map { |l| l.scan(/(?:^| )(.*?\.feature(?:(?::\d+)*))/) }.flatten
```
Comment (pasku) [inline lib/cucumber/cli/rerun_file.rb]: lines.flat_map { |l| l.scan(/(?:^| )(.*?\.feature(?:(?::\d+)*))/) }

http://ruby-doc.org/core-2.1.0/Enumerable.html#method-i-flat_map

Comment (mcrmfc) [inline lib/cucumber/cli/rerun_file.rb]: @pasku - agree that looks cleaner. The issue is this only flattens the arrays produced by `lines` and `scan`, but there are also captures which produce arrays and this doesn't deal with those (I think :confused:) ...can you think of anything else?

**Comment by johnwake** [comment]
Good work @mcrmfc :bow:

**Comment by tooky** [comment]
Thanks! @mcrmfc

**Pull request merged**